### PR TITLE
Add Mac Catalyst support

### LINKED ISSUE #2950: Add support for Catalyst

### PR DESCRIPTION
**Pull request by fealebenpae**
## Description
Add support for the `net6.0-maccatalyst` framework.

Resolves #2950 

##  TODO

* [x] Changelog entry
* [x] Tests (if applicable)

### CODE REVIEW

File: Tests/Tests.Maui/Tests.Maui.csproj [hunk 1]
```diff
@@ -69,6 +68,12 @@
       <SmartLink>False</SmartLink>
     </NativeReference>
   </ItemGroup>
+  <ItemGroup Condition="'$(UseRealmNupkgsWithVersion)' == ''">
+    <NativeReference Include="..\..\wrappers\build\macosx\$(Configuration)\realm-wrappers.framework" Condition="'$(TargetFramework)' == 'net6.0-maccatalyst'">
```
[the remark this answers is not in the document]
Comment (fealebenpae) [reply Tests/Tests.Maui/Tests.Maui.csproj:72]: This `NativeReference` only applies to in-solution builds where we use the locally-built framework. The zipped framework is used in the nuget package, where the tooling handles uncompressing it and injecting in the app bundle at the appropriate build stage.

File: wrappers/src/app_cs.cpp [hunk 2]
```diff
@@ -160,7 +160,7 @@ extern "C" {
                 sync_client_config.metadata_mode = app_config.metadata_mode;
             }
             else {
-#if REALM_PLATFORM_APPLE && !TARGET_OS_SIMULATOR
+#if REALM_PLATFORM_APPLE && !TARGET_OS_SIMULATOR && !TARGET_OS_MACCATALYST
```
Comment (nirinchev) [inline wrappers/src/app_cs.cpp:163]: So we don't support automatic encryption key management for catalyst?

File: Tests/Tests.Maui/Tests.Maui.csproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (nirinchev) [reply Tests/Tests.Maui/Tests.Maui.csproj:72]: Oh... so with the post_build command, we end up with both a `realm-wrappers.framework` and `realm-wrappers.framwork.zip`? And the tests projects uses the framework, while the nuget package contains the zip?

**Review by nirinchev — APPROVED** [review]
Looks good, just some clarification questions from me

File: Tests/Tests.Maui/Tests.Maui.csproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (fealebenpae) [reply Tests/Tests.Maui/Tests.Maui.csproj:72]: Yes, that’s exactly right.

**Pull request merged**